tensorly/backend/sparse_backend.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+import scipy.linalg
+import scipy.sparse.linalg
+
+# TODO: rehape, moveaxis, where, copy, transpose
+# TODO: arange, ones, zeros, zeros_like
+# TODO: kron
+# TODO: solve, qr
+from numpy import reshape, moveaxis, copy
+from numpy import arange, ones, zeros, zeros_like
+from numpy import kron
+from numpy import max, min, maximum, all, mean, sum, sign, abs, prod, sqrt
+from numpy.linalg import solve, qr
+import opt_einsum
+
+import sparse
+from .. import numpy_backend
+
+from sparse import concatenate, where
+
+
+assert_raises = np.testing.assert_raises
+assert_equal = np.testing.assert_equal
+assert_ = np.testing.assert_
+assert_array_equal = np.testing.assert_array_equal
+
+tensor = numpy_backend.tensor
+shape = numpy_backend.shape
+ndim = numpy_backend.ndim
+clip = numpy_backend.clip
+norm = numpy_backend.norm
+assert_array_almost_equal = numpy_backend.assert_array_almost_equal
+
+
+def dot(a, b):
+    res = sparse.dot(a, b)
+    return tensor(res)
+
+
+def transpose(x):
+    return x.T
+
+
+def context(tensor):
+    return {'dtype': tensor.dtype, 'coords': tensor.coords}
+
+
+def tensor(data, dtype=None, coords=None):
+    data = np.asanyarray(data)
+    if dtype is not None:
+        data = data.astype(dtype)
+    if isinstance(data, np.ndarray):
+        if data.dtype.kind == 'f':
+            i = data < np.finfo(data.dtype).resolution
+            data[i] = 0
+        return sparse.COO.from_numpy(data)
+    assert coords is not None
+    return sparse.COO(coords, data)
```

Comment:
I'm curious, what is this function supposed to do?  Does it ever get a sparse array?  If so then it looks like it converts it into a numpy array and then back out to sparse, which is probably something to avoid.

Reply:
Good catch. This function tries to convert to a sparse array, and I've added a catch for when it's a sparse array.